app/src/usb.c

```diff
@@ -34,8 +37,83 @@ static K_SEM_DEFINE(hid_sem, 1, 1);
 
 static void in_ready_cb(const struct device *dev) { k_sem_give(&hid_sem); }
 
+static void out_ready_cb(const struct device *dev) {
+    size_t report_length;
+    int rc = hid_int_ep_read(hid_dev, NULL, 0, &report_length);
+    if (rc != 0) {
+        LOG_ERR("Failed to read USB report length: %d", rc);
+        return;
+    }
+
+    uint8_t *report = k_malloc(report_length);
```

Comment:
Ideally, we should try to avoid dynamic `k_alloc` like this if at all avoidable.  Until we have more than one output report, just use a stack allocated array using `sizeof(struct zmk_hid_led_report)` for the length.

Reply:
Although we only have one report output, the device can still receive packets of arbitrary length and if these packets are not handled correctly they could cause inconsistency in the USB buffer.

Reply:
Are we sure the USB buffer stuff won't handle this? I'm not seeing anything in the docs.

Reply:
If we received a 3-byte report and read only the first 2 bytes, 1 byte would remain in the buffer. Doing so, on the next reading we would first read the previous byte remaining in the buffer causing incorrect behavior.